imports/models/BoardFavoritesCollection.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseCollection } from './BaseCollection'
+import { BoardRecord } from './BoardCollection'
+
+export interface BoardFavoriteRecord {
+  userid: string
+  boardId: string
+}
+
+export class BoardFavoritesCollection extends BaseCollection<BoardFavoriteRecord> {
+  constructor() {
+    super('board_favorites')
+  }
+
+  public async addFavoriteBoard(userid: string, boardId: string): Promise<void> {
+    this.collection.update({ userid, boardId }, { userid, boardId }, { upsert: true })
+  }
+
+  public async removeFavoriteBoard(userid: string, boardId: string): Promise<void> {
+    await this.remove({ userid, boardId })
+  }
+
+  public async getFavoriteBoards(userid: string) {
+    const favoriteBoardRecords = (await this.collection
+      .find({ userid })
+      .fetchAsync()) as BoardFavoriteRecord[]
+    const favoriteBoardIds = favoriteBoardRecords.map(
+      (record: BoardFavoriteRecord) => record.boardId
```

Comment:
BoardFavoriteRecord here is duplicate I think, typescript should could infer the types automatically here.

Reply:
use destruct syntax here to simplify map

```
favoriteBoards.map(({ boardId }) => boardId)
```